src/jsx.d.ts

```diff
@@ -0,0 +1,9 @@
+import {VNode, VNodeData} from './vnode';
+
+declare global {
+  namespace JSX {
+    interface IntrinsicElements {
+      [elemName: string]: VNodeData;
```

Comment:
On first look, I'm surprised that this is `VNodeData` and not `VNode`.

Reply:
yeah so the type of Element is VNode as described on line 5. IntrinsicElement defines what props an element accepts. so we say any dom element (including custom elements) can be created and it accepts the structure of VNodeData.

e.g ```
<input
   key='somekey' 
   attrs={{title: `some input`, type: `text`}}
   on={{input: this.handleInput, click: this.handleClick}}
   props={{value: 'someValue'}}
   class={{fancy: true}}
/>
```